### PR TITLE
context menus

### PR DESCRIPTION
**Pull request by Nyameliaaaa**
sex

### CODE REVIEW

File: src/classes/Client.ts [hunk 1]
```diff
@@ -15,8 +15,13 @@ import AI from './AI';
 import { version } from '../../package.json';
 import PhishingHandler from './handlers/PhishingHandler';
 
+interface applicationCommands {
+    chatInput: Collection<string, ChatInputCommand>;
+    contextMenu: Collection<string, ContextMenuCommand>;
+}
```
Comment (SuperchupuDev) [inline src/classes/Client.ts]: can you move this to /types/applicationCommand?

File: src/context/avatar.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import Embed from '@classes/Embed';
+import { CommandInteraction } from 'discord.js';
+import { ContextMenuCommandBuilder } from '@discordjs/builders';
+import { ApplicationCommandType } from 'discord-api-types';
+
+export async function run(
+    client: FluorineClient,
+    interaction: CommandInteraction<'cached'>
+) {
+    const member = interaction.options.getMember('user');
```
Comment (SuperchupuDev) [inline src/context/avatar.ts]:
```suggestion
import { ContextMenuInteraction } from 'discord.js';
import { ContextMenuCommandBuilder } from '@discordjs/builders';
import { ApplicationCommandType } from 'discord-api-types';

export async function run(
    client: FluorineClient,
    interaction: ContextMenuInteraction<'cached'>
) {
    const member = interaction.targetMember;
```

File: src/classes/Client.ts (hunk 1 above)
Comment (Nyameliaaaa) [reply src/classes/Client.ts]: idk

**Comment by Nyameliaaaa** [comment]
deploy is having a moment

**Comment by Nyameliaaaa** [comment]
add .con at the end to make the deployer load the context command

**Review by hkubus — DISMISSED** [review]
yeah

File: src/context/howgay.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    client: FluorineClient,
+    interaction: UserContextMenuInteraction
+): Promise<void> {
+    const member = interaction.targetMember;
+
+    const percent = ['478823932913516544', '348591272476540928'].includes(
+        member.user.id
+    )
+        ? 100
+        : hash(`${member.user.id}`) % 101;
```
Comment (SuperchupuDev) [inline src/context/howgay.ts]:
```suggestion
    const percent = ['478823932913516544', '348591272476540928'].includes(
        member.id
    )
        ? 100
        : hash(member.id) % 101;
```

Comment (Nyameliaaaa) [reply src/context/howgay.ts]: member.id isnt valid

**Review by hkubus — DISMISSED** [review]
I APro

File: src/types/applicationCommand.d.ts [hunk 4]
```diff
@@ -1,12 +1,30 @@
...
     dev?: boolean;
 }
+
+export interface ContextMenuCommand {
+    run: (client: FluorineClient, interaction: ContextMenuInteraction) => void;
+    data: ContextMenuCommandBuilder;
+    dev?: boolean;
+}
+
+export interface applicationCommands {
```
Comment (SuperchupuDev) [inline src/types/applicationCommand.d.ts]: change the naming to `ApplicationCommands`

File: src/classes/Client.ts [hunk 5]
```diff
@@ -55,19 +55,29 @@ export default class FluorineClient extends Client {
...
         this.cmds = new CommandHandler().loadCommands();
-        this.ai = new AI(this);
-        this.applicationCommands =
-            new ApplicationCommandHandler().loadCommands();
+
+        const _applicationCommands = new ApplicationCommandHandler();
+        this.applicationCommands = {
+            chatInput: _applicationCommands.loadChatInput(),
+            contextMenu: _applicationCommands.loadContextMenu()
+        };
```
Comment (SuperchupuDev) [inline src/classes/Client.ts]:
```suggestion
        const { loadChatInput, loadContextMenu } = new ApplicationCommandHandler();
        this.applicationCommands = {
            chatInput: loadChatInput(),
            contextMenu: loadContextMenu()
        };
```

File: src/context/howgay.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+export async function run(
+    client: FluorineClient,
+    interaction: UserContextMenuInteraction
+): Promise<void> {
+    const member = interaction.targetMember;
+
+    const percent = ['<@478823932913516544>', '<@348591272476540928>'].includes(
+        `<@${member.user.id}>`
+    )
```
Comment (SuperchupuDev) [inline src/context/howgay.ts]:
```suggestion
    const percent = ['478823932913516544', '348591272476540928'].includes(
        member.user.id
    )
```

Comment (Nyameliaaaa) [reply src/context/howgay.ts]: see 591b030fe5574f474409c8906065371ed993c303

Comment (SuperchupuDev) [reply src/context/howgay.ts]: in this case, `member` will always be a `GuildMember`, so there's no need to do this in the context menu

File: src/cmds/howgay.ts [hunk 7]
```diff
@@ -9,10 +11,18 @@ export async function run(
         return message.reply(
             client.language.get(message.guild.preferredLocale, 'HOWGAY_ARGS')
         );
-    const number = Math.floor(Math.random() * 100);
+
+    const thing = args.join(' ');
```
Comment (SuperchupuDev) [inline src/cmds/howgay.ts]:
```suggestion
    const thing =
        message.mentions.members.first() ?? args.join(' ');
```

File: src/cmds/howgay.ts [hunk 8]
```diff
@@ -9,10 +11,18 @@ export async function run(
...
+        thing.toString()
+    )
+        ? 100
+        : hash(thing.toString()) % 101;
+
     message.reply(
         client.language.get(message.guild.preferredLocale, 'HOWGAY', {
-            percent: number,
+            percent,
             thing: message.mentions.members.first() || args.join(' ')
```
Comment (SuperchupuDev) [inline src/cmds/howgay.ts]:
```suggestion
            thing
```

File: src/context/avatar.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+        .setLocaleTitle('AVATAR')
+        .setImage(member.displayAvatarURL({ dynamic: true, size: 512 }));
+    interaction.reply({ embeds: [embed] });
+}
+
+export const data = new ContextMenuCommandBuilder()
+    .setName('Avatar')
+    .setType(ApplicationCommandType.User);
+
+export const dev = true;
```
Comment (SuperchupuDev) [inline src/context/avatar.ts]:
```suggestion
```

File: src/context/howgay.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+            thing: member
+        })
+    );
+}
+
+export const data = new ContextMenuCommandBuilder()
+    .setName('How Gay')
+    .setType(ApplicationCommandType.User);
+
+export const dev = true;
```
Comment (SuperchupuDev) [inline src/context/howgay.ts]:
```suggestion
```

File: src/cmds/howgay.ts [hunk 11]
```diff
@@ -9,11 +11,19 @@ export async function run(
         return message.reply(
             client.language.get(message.guild.preferredLocale, 'HOWGAY_ARGS')
         );
-    const number = Math.floor(Math.random() * 100);
+
+    const thing = message.mentions.members.first() ?? args.join(' ');
```
Comment (SuperchupuDev) [inline src/cmds/howgay.ts]:
```suggestion
    const thing = message.mentions.users.first() ?? args.join(' ');
```

File: src/context/howgay.ts [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    const percent = ['478823932913516544', '348591272476540928'].includes(
+        member.user.id
+    )
+        ? 100
+        : hash(`<@${member.user.id}>`) % 101;
+
+    interaction.reply(
+        client.language.get(interaction.locale, 'HOWGAY', {
+            percent,
+            thing: member
```
Comment (SuperchupuDev) [inline src/context/howgay.ts]:
```suggestion
    const user = interaction.targetUser;

    const percent = ['478823932913516544', '348591272476540928'].includes(
        user.id
    )
        ? 100
        : hash(user.toString()) % 101;

    interaction.reply(
        client.language.get(interaction.locale, 'HOWGAY', {
            percent,
            thing: user
```

**Review by SuperchupuDev — CHANGES_REQUESTED** [review]
some bugs i noticed

**Review by SuperchupuDev — APPROVED** [review]

**Review by hkubus — APPROVED** [review]

**Comment by SuperchupuDev** [comment]
good job 😄

**Pull request merged**